### PR TITLE
Adding ASP.NET WebAPI tutorial with Cloud Endpoints

### PR DESCRIPTION
**Pull request by simonz130**
Tutorial for .NET developers using Cloud Endpoints.

Learn how to create a sample ASP.NET Web API service, deploy it to Google Cloud Platform (GCP) App Engine flexible environment, and use Cloud Endpoints for API management.

- [x] Tech writing / styleguide review - @chenopis   
- [x] Formatting review for publishing - @jmdobry

### CODE REVIEW

File: tutorials/exposing-asp.net-webapi-using-.net-core-with-cloud-functions/index.md [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+        gcloud app create --project --region=`YOUR_REGION`
+
+    Congrats! We just finished with cloud setup! Now it's time to deploy the API specification and API implementation.
+
+## Step 5 - Deploy API Specification to Google Endpoints
+
+1. Open `openapi.yaml` that was created in step 3 in VS Code.
+
+2. Add `host` declaration right before the `paths` section. The value should be `YOUR-PROJECT-ID.appspot.com` (replace `YOUR-PROJECT-ID` with your cloud project Id):
+    <pre>
```
Comment (jmdobry) [inline tutorials/exposing-asp.net-webapi-using-.net-core-with-cloud-functions/index.md]: For security reasons, all HTML and anything that is HTML-like is stripped during publishing. To get around this, move HTML and HTML-like snippets into their own files, and then in your tutorial link to the files. Here's an example of a tutorial that does this: https://github.com/GoogleCloudPlatform/community/tree/master/tutorials/appengine-dropwizard

I see at least two places in this tutorial where HTML-link stuff would be stripped out.

File: tutorials/exposing-asp.net-webapi-using-.net-core-with-cloud-endpoints/index.md [hunk 2]
```diff
@@ -0,0 +1,211 @@
+---
+title: Exposing ASP.NET Web API using .NET Core with Cloud Endpoints
+description: Learn how to create a sample ASP.NET Web API service, deploy it to Google Cloud Platform (GCP) App Engine flexible environment, and use Cloud Endpoints for API management.
+author: zeltser
+tags: App Engine, Cloud Endpoints, .NET Core, ASP.NET, API Gateway, Web API, WebAPI, Deploy, API Management
+date_published: 2017-04-01
+---
+# Create ASP.NET Web API using .NET Core with Cloud Endpoints
+
+This tutorial shows how to create simple API using ASP.NET Core Web API,
```
[the remark this answers is not in the document]
Comment (simonz130) [reply tutorials/exposing-asp.net-webapi-using-.net-core-with-cloud-endpoints/index.md]: Done

File: tutorials/exposing-asp.net-webapi-using-.net-core-with-cloud-endpoints/index.md [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+deploy that API to App Engine Flexible Environment, then using Cloud Endpoints to expose and monitor the API.
```
[the remark this answers is not in the document]
Comment (simonz130) [reply tutorials/exposing-asp.net-webapi-using-.net-core-with-cloud-endpoints/index.md]: done

File: tutorials/exposing-asp.net-webapi-using-.net-core-with-cloud-endpoints/index.md [hunk 4]
```diff
@@ -0,0 +1,211 @@
...
+
+## Step 1 - Setup your development environment
+
+We are going to write code using Visual Studio Code. Therefore, you should install the following:
```
[the remark this answers is not in the document]
Comment (simonz130) [reply tutorials/exposing-asp.net-webapi-using-.net-core-with-cloud-endpoints/index.md]: Fixed

File: tutorials/exposing-asp.net-webapi-using-.net-core-with-cloud-endpoints/index.md [hunk 5]
```diff
@@ -0,0 +1,211 @@
...
+
+* [.NET Core 2.0.0 SDK](https://www.microsoft.com/net/core)
+* [Visual Studio Code](https://code.visualstudio.com/)
+* [OmniSharp C# extension for Visual Studio Code](https://marketplace.visualstudio.com/items?itemName=ms-vscode.csharp)
+
+## Step 2 - Create ASP.NET Core Web API using VS Code
+
+1. Open a command line prompt and run the following commands:
```
[the remark this answers is not in the document]
Comment (simonz130) [reply tutorials/exposing-asp.net-webapi-using-.net-core-with-cloud-endpoints/index.md]: ok

File: tutorials/exposing-asp.net-webapi-using-.net-core-with-cloud-endpoints/index.md [hunk 6]
```diff
@@ -0,0 +1,211 @@
...
+
+        mkdir ExposeAPIWithEndpointsCore
+        cd ExposeAPIWithEndpointsCore
+        dotnet new webapi
+
+    This will create a template Web API project using .NET Core 2.0.
+
+2. Open `ExposeAPIWithEndpointsCore` folder you created in VS Code. It will look like this:
```
[the remark this answers is not in the document]
Comment (simonz130) [reply tutorials/exposing-asp.net-webapi-using-.net-core-with-cloud-endpoints/index.md]: ok

File: tutorials/exposing-asp.net-webapi-using-.net-core-with-cloud-endpoints/index.md [hunk 7]
```diff
@@ -0,0 +1,211 @@
...
+        endpoints_api_service:
+          name: [SERVICE_NAME]
+          rollout_strategy: managed
+
+    Replace `[SERVICE_NAME]` with corresponding value of previous command output.
+
+    By creating `app.yaml` with `endpoints_api_service` section, we effectively created the binding that will ensure that every incoming request to our API (specification is detailed in configuration with config_id) that will hit `YOUR-PROJECT-ID.appspot.com` (“name”), will be routed to our service.
+    Now you are almost ready to deploy API implementation into AppEngine Flex. In order to separate a project from its publish package in dotnet core, you should deploy the outcome of `dotnet publish` command. That output by default doesn’t contain `app.yaml` file we created earlier.
+
+3. Edit `“ExposeAPIWithEndpointsCore.csproj”` in VSCore so it always copies `app.yaml` into the publish directory. Add this section before `</Project>` in `ExposeAPIWithEndpointsCore.csproj`:
```
[the remark this answers is not in the document]
Comment (simonz130) [reply tutorials/exposing-asp.net-webapi-using-.net-core-with-cloud-endpoints/index.md]: ok

File: tutorials/exposing-asp.net-webapi-using-.net-core-with-cloud-endpoints/index.md [hunk 8]
```diff
@@ -0,0 +1,211 @@
...
+4. Deploy API implementation to AppEngine Flex.
+
+        dotnet publish
+        gcloud app deploy bin\debug\netcoreapp2.0\publish\app.yaml
+
+5. Test that your APIs are exposed properly. Open `https://[YOUR-PROJECT-ID].appspot.com/api/Values` in any browser. If you get **["value1","value2"]** back, you are done!
+
+## Congratulations!
+
+You now have the API deployed and exposed via Cloud Endpoints! In order to prove that, go to [Endpoints](https://console.cloud.google.com/endpoints) page of Google Cloud Console. There you can inspect the requests graph. It should reflect the requests you made to test the API. Also you can see all the deployments of endpoints configuration.
```
[the remark this answers is not in the document]
Comment (simonz130) [reply tutorials/exposing-asp.net-webapi-using-.net-core-with-cloud-endpoints/index.md]: rephrased

File: tutorials/exposing-asp.net-webapi-using-.net-core-with-cloud-endpoints/index.md [hunk 9]
```diff
@@ -0,0 +1,211 @@
...
+
+3. In Google SDK command prompt, make sure you are in the folder that contains `openapi.yaml` that was created in step 3. Execute:
+
+        gcloud endpoints services deploy openapi.yaml
+
+    You should notice that the deployment shows a few warnings, such as `Operation does not require an API Key`. You can ignore these for now.
+
+    So what has happened now? You just deployed API specification. This created a new Cloud Endpoints service configuration entry with the name that we specified in `“host”` field of `openapi.yaml`. This service entry is pre-configured to serve the API based on the specifications in yaml file. Now we want to ensure that all incoming API calls will be intercepted by Cloud Endpoints and then routed for execution to the service we deploy to AppEngine Flexible.
+
+## Step 6 - Deploy API implementation to AppEngine Flexible
```
[the remark this answers is not in the document]
Comment (simonz130) [reply tutorials/exposing-asp.net-webapi-using-.net-core-with-cloud-endpoints/index.md]: ok

File: tutorials/exposing-asp.net-webapi-using-.net-core-with-cloud-endpoints/index.md [hunk 10]
```diff
@@ -0,0 +1,211 @@
...
+5. Test that your APIs are exposed properly. Open `https://[YOUR-PROJECT-ID].appspot.com/api/Values` in any browser. If you get **["value1","value2"]** back, you are done!
+
+## Congratulations!
+
+You now have the API deployed and exposed via Cloud Endpoints! In order to prove that, go to [Endpoints](https://console.cloud.google.com/endpoints) page of Google Cloud Console. There you can inspect the requests graph. It should reflect the requests you made to test the API. Also you can see all the deployments of endpoints configuration.
+![Monitoring API](./MonitorAPI.png)
+
+## Summary
+
+We created a basic ASP.NET Core Web API, deployed it to AppEngine Flex on Google Cloud Platform and configured Cloud Endpoints in front of our API, so all the incoming requests will be sent to Cloud Endpoints which in turn will route them to our API implementation. We haven’t implemented all the non functional requirements of exposing the API like security, throttling and monitoring yet. In this tutorial our API is exposed to everyone without any authentication.
```
[the remark this answers is not in the document]
Comment (simonz130) [reply tutorials/exposing-asp.net-webapi-using-.net-core-with-cloud-endpoints/index.md]: I think summary is important - you can't have the API exposed without dealing with security at all. This is what the summary explains. However I also thinkn the What's next section makes sense. Added it.

**Comment by jmdobry** [comment]
Published: https://cloud.google.com/community/tutorials/exposing-aspnet-webapi-using-dotnetcore-with-cloud-endpoints

Please take a look and make sure it looks okay.

**Comment by simonz130** [comment]
The tutorial looks good. Thanks a lot for the help!
I'll publish the blog post that will point to it later today.

Thanks,
Simon

**Pull request merged**